.github/workflows/integration-pr.yml

```diff
@@ -0,0 +1,221 @@
+# **This is not a trusted workflow.**
+# **what?** - This workflow runs integration tests for supported OS,
+# python versions and core adapters. It will only run tests for
+# related file changes. Use the `test all` and `test ${adapter}`
+# label to manually run all or additional tests. Use `ok to test`
+# label to mark PRs from forked repositories that are safe to run integration
+# tests for. Requires secrets to run against different warehouses.
+# **why?** - This checks the functionality of dbt from a user's perspective,
+# provides code coverage, and attempts to catch functional regressions.
+# **when?** - This workflow will run for all pull requests (including those
+# forks).
+
+name: PR Integration Tests
+
+on:
+  pull_request_target:
+
+permissions: read-all
+
+concurrency:
+  group: integration-pr-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-checks:
+    runs-on: ubuntu-latest
+    outputs:
+      has-postgres-changes: ${{ steps.check-changes.outputs.postgres == 'true' }}
+      has-snowflake-changes: ${{ steps.check-changes.outputs.snowflake == 'true' }}
+      has-bigquery-changes: ${{ steps.check-changes.outputs.bigquery == 'true' }}
+      has-redshift-changes: ${{ steps.check-changes.outputs.redshift == 'true' }}
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Check if relevant files changed
+        uses: dorny/paths-filter@v2
+        id: check-changes
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          filters: |
+            postgres:
+              - 'core/**'
+              - 'plugins/postgres/**'
+            snowflake:
+              - 'core/**'
+              - 'plugins/snowflake/**'
+            bigquery:
+              - 'core/**'
+              - 'plugins/bigquery/**'
+            redshift:
+              - 'core/**'
+              - 'plugins/redshift/**'
+              - 'plugins/postgres/**'
```

Comment:
This is included because redshift code inherits a lot of things from the postgres implementation.